regulation-worker/internal/destination/destination.go

```diff
@@ -35,6 +58,38 @@ func (d *DestMiddleware) GetDestDetails(destID string) (model.Destination, error
 			}
 		}
 	}
+	if destDetail.Name == "" {
+		return model.Destination{}, model.ErrInvalidDestination
+	}
 
+	pkgLogger.Debugf("obtained destination detail: %v", destDetail)
 	return destDetail, nil
 }
+
+func (d *DestMiddleware) getDestDetails(ctx context.Context) (backendconfig.ConfigT, error) {
+	pkgLogger.Debugf("getting destination details with exponential backoff")
+
+	maxWait := time.Minute * 10
+	var err error
+	bo := backoff.NewExponentialBackOff()
+	boCtx := backoff.WithContext(bo, ctx)
+	bo.MaxInterval = time.Minute
+	bo.MaxElapsedTime = maxWait
+	var config backendconfig.ConfigT
+	var ok bool
+	if err = backoff.Retry(func() error {
+		config, ok = d.Dest.Get()
+		pkgLogger.Debugf("trying to update status...")
```

Comment:
Move the log up (before Get())
Correct the log statement. `Fetching backend config`.